File: lib/domain/cultivation_plan/errors/cultivation_plan_crop_missing_error.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Errors
      # 最適化時に計画に紐づく CultivationPlanCrop が欠落している（データ不整合）。
      class CultivationPlanCropMissingError < StandardError; end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/errors/cultivation_plan_crop_missing_error.rs
//! Ruby: `Domain::CultivationPlan::Errors::CultivationPlanCropMissingError`

use std::fmt;

#[derive(Debug, Clone, PartialEq)]
pub struct CultivationPlanCropMissingError {
    pub message: String,
}

impl CultivationPlanCropMissingError {
    pub fn new(message: impl Into<String>) -> Self {
        Self {
            message: message.into(),
        }
    }
}

impl fmt::Display for CultivationPlanCropMissingError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{}", self.message)
    }
}

impl std::error::Error for CultivationPlanCropMissingError {}
